src/plots/smith/smith.js

```diff
@@ -0,0 +1,1115 @@
+    } else {
+        xLength2 = yLength / arSector;
+        yLength2 = yLength;
+        gap = (xLength - xLength2) / gs.w / 2;
+        xDomain2 = [xDomain[0] + gap, xDomain[1] - gap];
+        yDomain2 = [yDomain[0], yDomain[1]];
+    }
+    _this.xLength2 = xLength2;
+    _this.yLength2 = yLength2;
+    _this.xDomain2 = xDomain2;
+    _this.yDomain2 = yDomain2;
+    // actual offsets from paper edge to the subplot box top-left corner
+    var xOffset2 = _this.xOffset2 = gs.l + gs.w * xDomain2[0];
+    var yOffset2 = _this.yOffset2 = gs.t + gs.h * (1 - yDomain2[1]);
+    // circle radius in px
+    var radius = _this.radius = xLength2 / dxSectorBBox;
+    // circle center position in px
+    var cx = _this.cx = xOffset2 - radius * sectorBBox[0];
+    var cy = _this.cy = yOffset2 + radius * sectorBBox[3];
+    // circle center in the coordinate system of plot area
+    var cxx = _this.cxx = cx - xOffset2;
+    var cyy = _this.cyy = cy - yOffset2;
+
+    _this.radialAxis = _this.mockAxis(fullLayout, smithLayout, radialLayout, {
+        _id: 'realaxis2',
+        side: {
+            counterclockwise: 'top',
+            clockwise: 'bottom'
+        }[radialLayout.side],
+        // keep track of real side
+        _realSide: radialLayout.side,
+        domain: [0, 100]
+    });
+
+    _this.angularAxis = _this.mockAxis(fullLayout, smithLayout, angularLayout, {
+        side: 'right',
+        // to get auto nticks right
+        domain: [0, Math.PI],
+        // don't pass through autorange logic
+        autorange: false
+    });
+
+    _this.doAutoRange(fullLayout, smithLayout);
+    // N.B. this sets _this.vangles
+    _this.updateAngularAxis(fullLayout, smithLayout);
+    // N.B. this sets _this.radialAxisAngle
+    _this.updateRadialAxis(fullLayout, smithLayout);
+    _this.updateRadialAxisTitle(fullLayout, smithLayout);
+
+    _this.xaxis = _this.mockCartesianAxis(fullLayout, smithLayout, {
+        _id: 'x',
+        domain: xDomain2
+    });
+
+    _this.yaxis = _this.mockCartesianAxis(fullLayout, smithLayout, {
+        _id: 'y',
+        domain: yDomain2
+    });
+
+    var dPath = _this.pathSubplot();
+
+    _this.clipPaths.forTraces.select('path')
+        .attr('d', dPath)
+        .attr('transform', strTranslate(cxx, cyy));
+
+    layers.frontplot
+        .attr('transform', strTranslate(xOffset2, yOffset2))
+        .call(Drawing.setClipUrl, _this._hasClipOnAxisFalse ? null : _this.clipIds.forTraces, _this.gd);
+
+    layers.bg
+        .attr('d', dPath)
+        .attr('transform', strTranslate(cx, cy))
+        .call(Color.fill, smithLayout.bgcolor);
+};
+
+proto.mockAxis = function(fullLayout, smithLayout, axLayout, opts) {
+    var ax = Lib.extendFlat({}, axLayout, opts);
+    setConvertSmith(ax, smithLayout, fullLayout);
+    return ax;
+};
+
+proto.mockCartesianAxis = function(fullLayout, smithLayout, opts) {
+    var _this = this;
+    var axId = opts._id;
+
+    var ax = Lib.extendFlat({type: 'linear'}, opts);
```

Comment:
Please remove `type: 'linear'` from here.

Reply:
When I remove it, the smith plot stops working. I am not sure why though--`ax.type` isn't really used anywhere in the code.